app/upgrades/v0.0.12.go

```diff
@@ -0,0 +1,130 @@
+
+			// Add service module `target_num_relays` parameter per `config.yml`.
+			// We don't use `GetParams()` to avoid potential protobuf issues and all networks have the same value (no need to read).
+			serviceParams := servicetypes.Params{
+				AddServiceFee: &cosmosTypes.Coin{
+					Denom:  "upokt",
+					Amount: math.NewInt(1000000000),
+				},
+				TargetNumRelays: 100,
+			}
+			err = keepers.ServiceKeeper.SetParams(ctx, serviceParams)
+			if err != nil {
+				logger.Error("Failed to set service params", "error", err)
+				return err
+			}
+			logger.Info("Successfully updated service params", "new_params", serviceParams)
+
+			// Add tokenomics module `global_inflation_per_claim` parameter per `config.yml`.
+			// We use GetParams() as `DaoRewardAddress` is different between networks and we don't want to hardcode it.
+			tokenomicsParams := keepers.TokenomicsKeeper.GetParams(ctx)
+			tokenomicsParams.GlobalInflationPerClaim = 0.1
+			err = keepers.TokenomicsKeeper.SetParams(ctx, tokenomicsParams)
+			if err != nil {
+				logger.Error("Failed to set tokenomics params", "error", err)
+				return err
+			}
+			return
+		}
+
+		// Returns the upgrade handler for v0.0.12
+		return func(ctx context.Context, plan upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+			logger := cosmosTypes.UnwrapSDKContext(ctx).Logger()
+			logger.Info("Starting v0.0.12 upgrade handler")
+
+			err := applyNewParameters(ctx)
+			if err != nil {
+				logger.Error("Failed to apply new parameters", "error", err)
+				return vm, err
+			}
+
+			// Since we changed the type of RevSharePercent from float32 to uint64, we need to update all on-chain data.
+			// The easiest way to do this is to iterate over all suppliers and services and set the revshare to 100 by default.
+			suppliers := keepers.SupplierKeeper.GetAllSuppliers(ctx)
+			logger.Info("Updating all suppliers to have a 100% revshare to the supplier", "num_suppliers", len(suppliers))
+			for _, supplier := range suppliers {
+				for _, service := range supplier.Services {
+					// Force all services to have a 100% revshare to the supplier.
+					// Not something we would do on a real mainnet, but it's a quick way to resolve the issue.
+					// Currently, we don't break any existing suppliers (as all of them have a 100% revshare to the supplier).
+					service.RevShare = []*sharedtypes.ServiceRevenueShare{
+						{
+							Address:            supplier.OperatorAddress,
+							RevSharePercentage: uint64(100),
+						},
+					}
+				}
+				keepers.SupplierKeeper.SetSupplier(ctx, supplier)
+				logger.Info("Updated supplier", "supplier", supplier.OperatorAddress)
+			}
+
+			logger.Info("Running module migrations")
+			vm, err = mm.RunMigrations(ctx, configurator, vm)
+			if err != nil {
+				logger.Error("Failed to run migrations", "error", err)
+				return vm, err
+			}
+
+			logger.Info("Successfully completed v0.0.12 upgrade handler")
```

Comment:
Let's make `v0.0.12` a local constant and do some string formatting here.

It'll avoid copy-pasta errors in the future.